### PR TITLE
Dynamically choose which certificates.k8s.io version to support

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
We had to update certificates API, now I'm returning the support for older certificates.k8s.io/v1beta1, but NOW it can be chosen based on API availability.

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
Dynamically choose which certificates.k8s.io version to support
```

### CODE REVIEW

File: pkg/controller/nodecsrapprover/node_csr_approver.go [hunk 1]
```diff
@@ -59,19 +61,72 @@ type reconciler struct {
...
 	if err != nil {
-		return fmt.Errorf("failed to create certificate client: %v", err)
+		return fmt.Errorf("failed to init discovery client: %w", err)
 	}
 
-	r := &reconciler{Client: mgr.GetClient(), certClient: certClient.CertificateSigningRequests()}
-	c, err := controller.New(ControllerName, mgr, controller.Options{Reconciler: r})
+	srvGroups, err := discoveryClient.ServerGroups()
+	if err != nil {
+		return fmt.Errorf("failed to server API groups: %w", err)
```
Comment (xmudrii) [inline pkg/controller/nodecsrapprover/node_csr_approver.go:73]: Tiny nit: it feels like this error message is missing a word.

```suggestion
		return fmt.Errorf("failed to get server API groups: %w", err)
```

File: pkg/controller/nodecsrapprover/node_csr_approver.go [hunk 2]
```diff
@@ -59,19 +61,72 @@ type reconciler struct {
...
+
+	switch certificatesVersionFound {
+	case "v1":
+		certClient, err := certificatesv1client.NewForConfig(mgr.GetConfig())
+		if err != nil {
+			return fmt.Errorf("failed to create certificate client: %v", err)
+		}
+		rec = &reconciler{Client: mgr.GetClient(), certClient: certClient.CertificateSigningRequests()}
+		watchType = &certificatesv1.CertificateSigningRequest{}
+	case "v1beta1":
```
Comment (xmudrii) [inline pkg/controller/nodecsrapprover/node_csr_approver.go:106]: Is it expected that we don't assign `watchType` for v1beta1?

File: pkg/controller/nodecsrapprover/node_csr_approver.go (hunk 1 above)
Comment (kron4eg) [reply pkg/controller/nodecsrapprover/node_csr_approver.go:73]: indeed :D

File: pkg/controller/nodecsrapprover/node_csr_approver.go (hunk 2 above)
Comment (kron4eg) [reply pkg/controller/nodecsrapprover/node_csr_approver.go:107]: good catch, no, it was forgotten

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**